File: src/employeemanager/model/Instructor.java
package employeemanager.model;

import employeemanager.exceptions.InvalidDataException;

import java.time.LocalDate;

public class Instructor extends FullTimeEmployee {

    public Instructor(int employeeId,
                      String firstName,
                      String lastName,
                      LocalDate startDate,
                      LocalDate dateOfBirth,
                      double annualSalary,
                      int vacationDays,
                      String benefits) throws InvalidDataException {
        super(employeeId, firstName, lastName, startDate, dateOfBirth,
              annualSalary, vacationDays, benefits);
    }

    public void teach() {
        System.out.println(getFullName() + " is teaching a class.");
    }

    @Override
    public void reportToManager() {
        System.out.println("[Instructor] " + getFullName() + " reports to department chair.");
    }
}

File: Part 3/EmployeeManager/models/Instructor.cs
using System;

namespace EmployeeManager.Models
{
    public class Instructor : FullTimeEmployee
    {
        public Instructor(int employeeId, string firstName, string lastName, DateTime startDate, DateTime dateOfBirth, double annualSalary, int vacationDays, string benefits) : base(employeeId, firstName, lastName, startDate, dateOfBirth, annualSalary, vacationDays, benefits) { }

        public void Teach() => Console.WriteLine($"{FullName} is instructing a class");

        public override void ReportToManager() => Console.WriteLine($"[Instructor] {FullName} reports to the department chair");
    }
}